### ISSUE TITLE
tf.layers.Layer does not preserver the keras metadata entered into a tensor/placeholder

### ISSUE THREAD

**Issue by kirk86** [tensorflow/tensorflow#19292]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.8.0-rc1
- **Python version**:  3.6
- **Bazel version (if compiling from source)**: 0.12.0
- **GCC/Compiler version (if compiling from source)**: 5.4.1
- **CUDA/cuDNN version**: 7.1
- **GPU model and memory**: 1080Ti
- **Exact command to reproduce**:  Any command like `tf.keras.layers.TimeDistributed(tf.layers.Conv2D(64, 3, padding='valid', activation=tf.nn.relu)(x))` where `x` is any tensor that has keras metadata like `_keras_shape` and/or `_keras_history`,

### Describe the problem
 wrapping any tensor which contains keras metadata such as `_keras_shape` or `_keras_histroy` with `tf.layers.Layer` or passing it through `tf.layers.Conv2D` does not preserve the keras metadata instead it discards them. In addition the `tf.layers.Layer` does not implement any `_add_inbound_node` method which is really confusing and leads to many errors with shape mismatches.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 16 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 32 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by kirk86** [comment]
Yes it is still an issue!

**Comment by bignamehyp** [comment]
@pavithrasv can you please take a look to see if this is by designed or a bug?

**Comment by pavithrasv** [comment]
@kirk86 tf.layers can be used the same way as tf.keras layers and they will maintain the keras metadata.  For example, you can do:

```
      inputs = keras.Input(shape=(10,))
      x = tf_core_layers.Dense(32, activation=nn.relu)(inputs)
      outputs = tf_core_layers.Dense(2, activation=nn.softmax)(x)
      model = keras.Model(inputs, outputs)
```
You are seeing an issue because of the way TimeDistributed layer has been initialized. TimeDistributed should be initialized with a `Layer` and not a tensor. It applies the wrapped layer to every timestep in the input sequences. 

```
inputs = Input(shape=(5, 4)) # 5 => 5 timesteps
output = TimeDistributed(Dense(2))(inputs) # output of this layer will be (5, 2)
```
Please try after changing your code as follows:
```
tf.keras.layers.TimeDistributed(tf.layers.Conv2D(64, 3, padding='valid', activation=tf.nn.relu))(x) 
```
We will improve the error message for this layer so that it is easier to identify the issue.
